projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -225,12 +238,73 @@ export class DatagridComponent<R> implements OnInit {
     ngOnInit(): void {
         this.isLoading = true;
         this.gridRefresh.emit({});
+        this.updateSeletionInformation();
+    }
+
+    private updateSelectedItems(): void {
+        if (this._selectionType === GridSelectionType.Single) {
+            console.log(this.datagrid.selection.currentSingle);
+            let found = false;
+            this.items.forEach(
+                (item, itemIndex) =>
+                    (found =
+                        found ||
+                        this.trackBy(itemIndex, item) ===
+                            this.trackBy(
+                                this.items.indexOf(this.datagrid.selection.currentSingle),
+                                this.datagrid.selection.currentSingle
+                            ))
+            );
+            if (!found) {
+                this.datagrid.selection.currentSingle = undefined;
+            }
+        } else if (this._selectionType === GridSelectionType.Multi) {
+            if (this.datagrid.selection.current) {
+                this.datagrid.selection.current = this.datagrid.selection.current.filter((selected, selectedIndex) => {
+                    let found = false;
+                    this.items.forEach(
+                        (item, itemIndex) =>
+                            (found = found || this.trackBy(itemIndex, item) === this.trackBy(selectedIndex, selected))
+                    );
+                    return found;
+                });
+            }
+        }
+    }
+
+    private updateSeletionInformation(): void {
+        if (!this.datagrid) {
+            return;
+        }
+        if (this._selectionType === GridSelectionType.Single) {
+            this.datagrid.selected = undefined;
+            this.datagrid.singleSelected = this.singleSelected;
+        } else if (this._selectionType === GridSelectionType.Multi) {
+            this.datagrid.singleSelected = undefined;
+            this.datagrid.selected = this.multiSelection;
+        } else if (this._selectionType === GridSelectionType.None) {
+            this.datagrid.selected = [];
+            this.datagrid.singleSelected = undefined;
+            this.datagrid.selected = undefined;
+        }
+    }
+
+    /**
+     * Returns the items selected in the VCD datagrid.
+     */
+    getDatagridSelection(): R[] {
+        if (this.datagrid.selection.currentSingle !== undefined && this.datagrid.selection.currentSingle !== null) {
```

Comment:
Why check for `undefined` explicitly?
`if (this.datagrid.selection.currentSingle)`? Not sure why the extra checks for `null` and undefined. It's not like it could ever be 0 or an empty string